create_patched_sdk.sh

```diff
@@ -1,182 +1,233 @@
+	exit 0
+fi
+
+sdk_platform="$7"
+if [[ $# -lt 7 ]] || ignored $sdk_platform; then
+	echo "No sdk_platform provided. Defaulting to iOS."
+	sdk_platform="iOS"
+elif [[ $(echo $sdk_platform | tr '[A-Z]' '[a-z]') == ios ]]; then
+	sdk_platform="iOS"
+elif [[ $(echo $sdk_platform | tr '[A-Z]' '[a-z]') == tvos ]]; then
+	sdk_platform="tvOS"
+else
+	echo "Unsupported sdk_platform provided. Please choose either 'iOS' or 'tvOS'."
+	exit 1
 fi
 
 sdks_output_path="$2"
 if [[ $# -lt 2 ]] || ignored $sdks_output_path; then
-    if [[ -z $THEOS ]]; then
-        printf 'No Theos installation found. Please either install Theos or provide a path to an sdks directory\n\n'
-        print_usage
-
-        exit 1
-    fi
-    sdks_output_path="$THEOS/sdks"
+	if [[ -z $THEOS ]]; then
+		printf 'No Theos installation found. Please either install Theos or provide a path to an sdks directory\n\n'
+		print_usage
+		exit 1
+	fi
+	sdks_output_path="$THEOS/sdks"
 fi
 
 if ! [[ -d $sdks_output_path ]]; then
-    mkdir -p "$sdks_output_path"
+	mkdir -p "$sdks_output_path"
 fi
 
 # tbd info
 version="v3"
-
-archs_option=("--replace-archs" armv7 armv7s arm64 arm64e)
-tbd_options=("--ignore-clients" "--ignore-undefineds" "--allow-private-objc-symbols" "--ignore-missing-exports")
-write_options=("--preserve-subdirs" "--replace-path-extension")
+if [[ $sdk_platform == iOS ]]; then
+	archs_option=("--replace-archs" armv7 armv7s arm64 arm64e)
+	tbd_options=("--ignore-clients" "--ignore-undefineds" "--allow-private-objc-symbols" "--ignore-missing-exports")
+	write_options=("--preserve-subdirs" "--replace-path-extension")
+else
+	archs_option=("--replace-archs" arm64)
+	tbd_options=("--allow-private-objc-symbols" "--ignore-missing-exports")
+	#write_options=("--maintain-directories" "--replace-path-extension")
+	write_options=("--preserve-subdirs")
+fi
 
 no_overwrite="--no-overwrite"
 if [[ $# -gt 2 ]] && ! ignored $3; then
-    no_overwrite=""
+	no_overwrite=""
 fi
 
-no_warnings="--ignore-warnings"
+no_warnings=""
 tbd_tool="$5"
+if [[ $# -gt 3 ]] && ! ignored $4; then
+	no_warnings="--ignore-warnings"
+fi
 
 if [[ $# -lt 5 ]] || ignored $tbd_tool; then
-    tbd_tool="tbd"
-    tbd_exists=$(command -v $tbd_tool)
-
-    if [[ -z $tbd_exists ]]; then
-        printf 'No installation of tbd found. Please install the latest release of tbd from here; https://github.com/inoahdev/tbd/releases or provide a path to a tbd installation\n\n'
-        print_usage
-
-        exit 1
-    fi
+	tbd_tool="tbd"
+	if [[ -x $(which $tbd_tool) ]]; then
+		printf 'No installation of tbd found. Please install the latest release of tbd from here; https://github.com/inoahdev/tbd/releases or provide a path to a tbd installation\n\n'
+		print_usage
+		exit 1
+	fi
 else
-    tbd_exists=$(command -v "$tbd_tool")
-    if [[ -z $tbd_exists ]]; then
-        printf "Provided tbd-tool (%s) doesn't exist or isn't executable\n" "$tbd_tool"
-        exit 1
-    fi
+	if [[ -x $(which $tbd_tool) ]]; then
```

Comment:
this too, I think should be negated
```suggestion
	if ! [[ -x $(which $tbd_tool) ]]; then
```